pkg/notification/manager.go

```diff
@@ -66,6 +72,8 @@ func (am *AtomicManager) Remove(id int) {
 
 // Send sends the notification to the registered handlers
 func (am *AtomicManager) Send(notification interface{}) {
+	am.lock.Lock()
+	defer am.lock.Unlock()
 	for _, handler := range am.handlers {
 		handler(notification)
 	}
```

Comment:
Per our group discussion, we can first create an array of handlers, outside of the map, then trigger the handlers. That way we are not potentially modifying the Map while we're iterating.